### PR TITLE
Hosting Command Palette: Add/ssh commands 

### PR DESCRIPTION
**Pull request by katinthehatsite**
Related to https://github.com/Automattic/dotcom-forge/issues/4470

## Proposed Changes

This PR adds SSH commands that are mentioned in the issue above as indicated in the issue above. The new commands that are added are:

* Reset SSH/SFTP password (resets SFTP passwords and copies it to clipboard)
* Copy SSH connection string
* Open SSH details (redirects to Hosting Configuration page on AT sites)

## Testing Instructions

* Navigate to Calypso Live link or pull this branch
* Go to `/sites` page
* Press `cmd + k` on Mac or `ctrl + k` on Windows
* Search for `ssh`
* Try the SSH commands above and confirm you the data copied to the clipboard

Screenshot 2023-11-10 at 11 53 26 AM

**Notes**:

* At the moment, the command palette shows all the commands. I am wondering if we need to limit the number and display only some commands by default
* I am making an API request in `Copy SSH connection string` and `Reset SSH/SFTP password` to get the data that I need. It seems to be working relatively fast for me and it is the best approach that I found at the moment, but happy to reconsider if you think it leads to slowdown 
* Not sure if we need to limit the `Open SSH details` command only to Atomic sites or should keep it available for the sites on the Business plan

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by katinthehatsite** [comment]
@sejas I changed the commands as per: https://github.com/Automattic/dotcom-forge/issues/4470

I have a question about how we would want to handle errors e.g. if something goes wrong with resetting a password, what do we want to do in the UI? a notice in the top right corner and redirect to Hosting Configuration? Or just redirect to Hosting Configuration?

File: client/components/command-pallette/use-command-pallette.tsx [hunk 1]
```diff
@@ -69,18 +67,13 @@ export const useCommandPallette = ( {
...
-		ref: 'topbar',
-	} );
 	const { data: allSites = [] } = useSiteExcerptsQuery(
 		[],
 		( site ) => ! site.options?.is_domain_only
 	);
 
 	// Call the generateCommandsArray function to get the commands array
-	const commands = generateCommandsArrayWpcom( { setSelectedCommandName, __, createSiteUrl } );
+	const commands = useCommandsArrayWpcom( { setSelectedCommandName } );
```
Comment (sejas) [inline client/components/command-pallette/use-command-pallette.tsx:76]: Thanks for this refactor 🙏

File: client/sites-dashboard/components/wpcom-smp-commands.tsx [hunk 2]
```diff
@@ -24,6 +23,68 @@ export const generateCommandsArrayWpcom = ( {
...
+		siteId: number,
+		copyType: 'username' | 'connectionString',
+		siteSlug: string
+	) => {
+		const sshUser = await fetchSshUsersDetails( siteId, copyType );
+
+		if ( sshUser !== null ) {
+			navigator.clipboard.writeText( sshUser );
+		} else {
+			navigate( `/hosting-config/${ siteSlug }` );
```
Comment (sejas) [inline client/sites-dashboard/components/wpcom-smp-commands.tsx]: Nice fallback ✨

**Review by sejas — APPROVED** [review]
Great work Kat 👏 !!

I moved the logic around `copyType` outside of the fetch function. You can see the changes here https://github.com/Automattic/wp-calypso/pull/83964/commits/84e9d4d486d06422d038ecd8e44c5b73797f4b48

I added success notices https://github.com/Automattic/wp-calypso/pull/83964/commits/546d9b026e816abcccc214915300d0369bb747c8 , I wasn't sure if we needed them but after trying  it in live I found I prefer to add a confirmation for commands that do not navigate to other screen.

I'll deploy it when tests pass! Thank you!!


## Screencast

**Success**


https://github.com/Automattic/wp-calypso/assets/779993/dfab2845-78b7-4070-b33d-bc1995f63a2f



**Error**


https://github.com/Automattic/wp-calypso/assets/779993/d070d622-0a91-446a-b330-b854215aa15a

**Comment by sejas** [comment]
> A notice in the top right corner and redirect to Hosting Configuration? Or just redirect to Hosting Configuration?

Probably a notice will be a perfect fit for errors. But let's discuss it next week.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9666166

Some locales (Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @katinthehatsite for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9666166

Some locales (Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @katinthehatsite for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9666166

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @katinthehatsite for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9666166).

**Pull request merged**